Celeste.Mod.mm/Patches/LevelLoader.cs

```diff
@@ -190,8 +203,38 @@ public patch_LevelLoader(Session session, Vector2? startPosition = default)
                 LoadingThread();
             } catch (Exception e) {
                 string sid = session?.Area.GetSID() ?? "NULL";
-                patch_LevelEnter.ErrorMessage = Dialog.Get("postcard_levelloadfailed").Replace("((sid))", sid);
-                Logger.Log(LogLevel.Warn, "LevelLoader", $"Failed loading map {sid}");
+                if (patch_LevelEnter.ErrorMessage == null) {
+                    if (e is KeyNotFoundException && e.StackTrace.Contains("TileHandler") && e.Data.Count != 0) {
+                        int x = (int)e.Data["x"];
```

Comment:
would it be good to check if these keys are present?

Reply:
`Exception.Data` is explicitly for user-defined purposes, and currently this is the only place Everest uses it (alternative was embedding data in the exception message + messy regex). If we expand exception handling we can add more checks but for now I think I prefer it cleaner